src/libraries/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -259,16 +259,68 @@ private void Resize()
         }
     }
 
+    internal sealed class CollectionLookup<TKey, TElement> : Lookup<TKey, TElement>, ICollection<IGrouping<TKey, TElement>>, IReadOnlyCollection<IGrouping<TKey, TElement>>
```

Comment:
These types of optimization seem to fuzzy the distinction between `.SpeedOpt.cs` files and the main implementation. Is the separation still needed?

Reply:
The separation is used for browser, where it was previously decided that all of the extra impls en mass were significantly increasing wasm size and compilation times. I wouldn't want to remove them all wholesale without revisiting the concern, though it'd be nice eventually to not have the different flavors, or to make it a trimming decision via switches.